content/en/blog/2024/authz-policy-with-kyverno/index.md

```diff
@@ -0,0 +1,469 @@
+- The `istioctl` command-line tool installed.
+
+Install Istio and configure your [mesh options](/docs/reference/config/istio.mesh.v1alpha1/) to enable Kyverno:
+
+{{< text bash >}}
+$ istioctl install -y -f - <<EOF
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  meshConfig:
+    accessLogFile: /dev/stdout
+    accessLogFormat: |
+      [KYVERNO DEMO] my-new-dynamic-metadata: '%DYNAMIC_METADATA(envoy.filters.http.ext_authz)%'
+    extensionProviders:
+    - name: kyverno-authz-server.local
+      envoyExtAuthzGrpc:
+        service: kyverno-authz-server.kyverno.svc.cluster.local
+        port: '9081'
+EOF
+{{< /text >}}
+
+Notice that in the configuration, we define an `extensionProviders` section that points to the Kyverno Authz Server installation:
+
+{{< text yaml >}}
+[...]
+    extensionProviders:
+    - name: kyverno-authz-server.local
+      envoyExtAuthzGrpc:
+        service: kyverno-authz-server.kyverno.svc.cluster.local
+        port: '9081'
+[...]
+{{< /text >}}
+
+#### Deploy the Kyverno Authz Server
+
+The Kyverno Authz Server is a GRPC server capable of processing Envoy External Authorization requests.
+
+It is configurable using Kyverno `AuthorizationPolicy` resources, either stored in-cluster or provided externally.
+
+{{< text bash >}}
+$ kubectl create ns kyverno
+{{< /text >}}
+
+{{< text bash >}}
+$ kubectl label namespace kyverno istio-injection=enabled
+{{< /text >}}
+
+{{< text bash >}}
+$ helm install kyverno-authz-server --namespace kyverno --wait --repo https://kyverno.github.io/kyverno-envoy-plugin kyverno-authz-server
+{{< /text >}}
+
+#### Deploy the sample application
+
+Httpbin is a well-known application that can be used to test HTTP requests and helps to show quickly how we can play with the request and response attributes.
+
+{{< text bash >}}
+$ kubectl create ns my-app
+{{< /text >}}
+
+{{< text bash >}}
+$ kubectl label namespace my-app istio-injection=enabled
+{{< /text >}}
+
+{{< text bash >}}
+$ kubectl apply -f {{< github_file >}}/samples/httpbin/httpbin.yaml -n my-app
+{{< /text >}}
+
+#### Deploy an Istio AuthorizationPolicy
+
+An `AuthorizationPolicy` defines the services that will be protected by the Kyverno Authz Server.
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: security.istio.io/v1
+kind: AuthorizationPolicy
+metadata:
+  name: my-kyverno-authz
+  namespace: istio-system # This enforce the policy on all the mesh being istio-system the mesh config namespace
+spec:
+  selector:
+    matchLabels:
+      ext-authz: enabled
+  action: CUSTOM
+  provider:
+    name: kyverno-authz-server.local
+  rules: [{}] # Empty rules, it will apply to selectors with ext-authz: enabled label
+EOF
+{{< /text >}}
+
+Notice that in this resource, we define the Kyverno Authz Server `extensionProvider` you set in the Istio configuration:
+
+{{< text yaml >}}
+[...]
+  provider:
+    name: kyverno-authz-server.local
+[...]
+{{< /text >}}
+
+#### Label the app to enforce the policy
+
```

Comment:
put a small description here of what you're doing?  Would you not normally just use `kubectl label` in this case?

Reply:
> Would you not normally just use kubectl label in this case?

Not really because we are adding the label to the pod template, not to the deployment.